tests/testthat/test-Lint-builder.R

```diff
@@ -2,26 +2,32 @@ test_that("Lint() errors on invalid input", {
   dummy_line <- "abc"
   expect_error(
     Lint("dummy.R", line = dummy_line, column_number = NA_integer_),
-    rex::rex("`column_number` must be an integer between 0 and nchar(line) + 1 (4). It was NA.")
+    "`column_number` must be an integer between 0 and `nchar(line) + 1` (4)",
+    fixed = TRUE
   )
   expect_error(
     Lint("dummy.R", line = dummy_line, line_number = 0L),
-    rex::rex("`line_number` must be a positive integer. It was 0.")
+    "`line_number` must be a positive integer",
+    fixed = TRUE
   )
   expect_error(
     Lint("dummy.R", ranges = c(1L, 3L)),
-    rex::rex("`ranges` must be NULL or a list.")
+    "`ranges` must be `NULL` or a <list>",
+    fixed = TRUE
   )
   expect_error(
     Lint("dummy.R", ranges = list(1L)),
-    rex::rex("`ranges` must only contain length 2 integer vectors without NAs.")
+    "`ranges` must only contain <integer> vectors of length 2 without `NA`s.",
```

Comment:
this looks really fragile TBH. What happens when {cli} changes the formatting of any of the `{.cls}` `{.fn}` `{.str}` etc markers? Maybe we need a helper like `cli_style()` that can match what {cli} does to the strings? Does {cli} itself offer anything like this?

Reply:
I completely agree with you that this makes the tests fragile, and I can see why the tidyverse projects use snapshots for testing exceptions.

I think the good news here is that, because a lot of revisions in `{cli}` have already happened based on the feedback it received during its adoption across many tidyverse/r-lib repos, its formatting API is quite stable. Thus, it is quite unlikely that the formatting for any of these markers will change because that will lead to test failures across many repositories. 

Should we find some comfort in that knowledge? Or do you think that would instead be complacency?

Reply:
>  it is quite unlikely that the formatting for any of these markers will change because that will lead to test failures across many repositories.

that's not quite true because snapshots are still not tested on CRAN, so breakages are not reported there. the volume of breakages of dev packages is not (AFAIK) monitored. we could ignore breakages in dev commit-by-commit and update snapshots as part of release, at the cost of not noticing real breakages when they happen.

Reply:
Most of the repos in question are under active development, and CI run snapshot tests almost on a daily basis as part of each build.
 
So, yes, if `{cli}` were to change its formatting API, there would be a big spike in failed CI builds across many repos.

I am not advocating for snapshot tests, but I am not too concerned about fragility of tests due to the changes in `{cli}` formatting either.

Reply:
@MichaelChirico Can I resolve this comment thread? Or do you have further thoughts here?

Reply:
are you ok being responsible with another pass over the tests to remove the cli formatting if we do get hit by such a breaking change? note that it would likely mean CRAN failing so would need a relatively fast turnaround time.

Reply:
Yes, I can take responsibility of updating the tests in case of formatting changes in `{cli}`.